certbot/tests/auth_handler_test.py

```diff
@@ -95,23 +94,24 @@ def _test_name1_tls_sni_01_1_common(self, combos):
         authzr = gen_dom_authzr(domain="0", challs=acme_util.CHALLENGES, combos=combos)
         mock_order = mock.MagicMock(authorizations=[authzr])
 
-        with mock.patch("certbot.auth_handler.AuthHandler._poll_challenges") as mock_poll:
-            mock_poll.side_effect = self._validate_all
+        self.mock_net.poll.side_effect = _gen_mock_on_poll(retry=1)
+        with mock.patch('certbot.auth_handler.time') as mock_time:
             authzr = self.handler.handle_authorizations(mock_order)
 
-        self.assertEqual(self.mock_net.answer_challenge.call_count, 1)
+            self.assertEqual(self.mock_net.answer_challenge.call_count, 1)
 
-        self.assertEqual(mock_poll.call_count, 1)
-        chall_update = mock_poll.call_args[0][1]
-        self.assertEqual(list(six.iterkeys(chall_update)), [0])
-        self.assertEqual(len(chall_update.values()), 1)
+            self.assertEqual(self.mock_net.poll.call_count, 2)  # Because there is one retry
+            self.assertEqual(mock_time.sleep.call_count, 2)
+            # Retry-After header is 1 second, but at the time sleep is invoked, several
+            # instructions are executed, and next pool is in less than a second.
+            self.assertTrue(mock_time.sleep.call_args_list[0][0][0] <= 1)
```

Comment:
It's probably pretty unlikely, but I'm worried about on a machine with limited resources, we could actually take longer than a second here and spuriously fail this test. Can we add an optional parameter to `_gen_mock_on_poll` to set the `Retry-After` value, set it to something fairly large here and then check it is less than or equal to that value but greater than 3?

If this is done, the comment in the code here will need to be updated as well.

Reply:
Done. And in fact I was not checking the correct time call. Corrected also.